packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -98,54 +103,8 @@ class HotRunner extends ResidentRunner {
     benchmarkData[name].add(value);
   }
 
-  Future<bool> _refreshDartDependencies() async {
-    if (!hotRunnerConfig.computeDartDependencies) {
-      // Disabled.
-      return true;
-    }
-    if (_dartDependencies != null) {
-      // Already computed.
-      return true;
-    }
-
-    try {
-      // Will return immediately if pubspec.yaml is up-to-date.
-      await pubGet(
```

Comment:
What will happen now if user changes pubspec.yaml in application they hot reload?

Reply:
We currently run a pub get when starting via flutter run or attach. It's not possible to actually update dependencies via pub and hot reload, because this changes .packages.

Reply:
If changing `.packages` doesn't work that I would imagine is also a bug. Once that is fixed(not sure where the culprit is), without this `pub get` step, changes to `pubspec.yaml` have no chance of being picked up though.

Reply:
Filled https://github.com/flutter/flutter/issues/29186